rsocket/statemachine/RSocketStateMachine.cpp

```diff
@@ -681,15 +703,18 @@ void RSocketStateMachine::handleUnknownStream(
   // TODO: comparing string versions is odd because from version
   // 10.0 the lexicographic comparison doesn't work
   // we should change the version to struct
-  if (frameSerializer_->protocolVersion() > ProtocolVersion{0, 0} &&
-      !streamsFactory_.registerNewPeerStreamId(streamId)) {
-    return;
+  if(frameType != FrameType::PAYLOAD) {
+    // don't check registerNewPeerStreamId if it's a payload - it may be an
+    // additional fragment
+    if (frameSerializer_->protocolVersion() > ProtocolVersion{0, 0} &&
+        !streamsFactory_.registerNewPeerStreamId(streamId)) {
+      return;
+    }
   }
 
-  if (!isNewStreamFrame(frameType)) {
+  if (!isNewStreamFrame(frameType) && !(frameType == FrameType::PAYLOAD)) {
```

Comment:
Processing PAYLOAD frames for unknown streams is allowed because it's possible that the payload is a fragment for a stream that hasn't been initialized yet. If it's not found in the stream cache, the connection is closed with an error.